### PR TITLE
[BugFix] Make FlashInferMetadataBuilder non-blocking

### PR DESCRIPTION
**Pull request by nvjullin**
## Purpose
The blocking H2D memcpys breaks overlap scheduler #23569, setting them to non-blocking fixes it.
The correctness is ensured by vllm/v1/worker/gpu_model_runner.py:2112
```python3
            if self.prepare_inputs_event is not None:
                # Ensure prior step has finished with reused CPU tensors.
                self.prepare_inputs_event.synchronize()
            try:
                # Prepare the decoder inputs.
                (attn_metadata, logits_indices, spec_decode_metadata,
                 num_scheduled_tokens_np, spec_decode_common_attn_metadata,
                 max_query_len, ubatch_slices, num_tokens_after_padding
                 ) = self._prepare_inputs(scheduler_output)
```

## Test Result

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

### CODE REVIEW

**Comment by nvpohanh** [comment]
@benchislett Please check if this fix is correct. Thanks!

**Review by mgoin — APPROVED** [review]
Looks reasonable to me, but @LucasWilkinson or @benchislett should validate before merge

**Review by LucasWilkinson — APPROVED** [review]
LGTM! Thanks for the contribution!
